lib/rspec/core/output_wrapper.rb

```diff
@@ -3,17 +3,25 @@ module Core
     # @private
     class OutputWrapper
       # @private
-      attr_writer :output
+      attr_accessor :output
 
       # @private
       def initialize(output)
         @output = output
       end
 
+      def respond_to?(name, priv=false)
+        output.respond_to?(name, priv)
+      end
+
+      def method_missing(name, *args, &block)
+        output.method_missing(name, *args, &block)
+      end
```

Comment:
I initially did exactly what you propose, but because of the wackyness with `deprecation_stream` not liking being an output wrapper abandoned it in favour of restoring the old behaviour.